### PR TITLE
Improves the "Sync Apps" panel UI. Closes #2698

### LINKED ISSUE #2698: Provide a progress meter for the Sync App panel
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

The new sync panel only syncs apps when the page is open, but you could refresh the page and it would stop.

**To Reproduce**

Create some apps and use the "sync apps" panel to sync. Reload the page. See that sync has stopped.

**Expected behavior**

The user should be discouraged from closing the page until the sync is complete.

**Version**

```shell
v1.3
```

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Closes #2698

Adds a progress meter when you're syncing multiple apps in the UI. This is NOT cancelable, but at the end you are presented with a "Close" button.

### Screenshots

Screen Shot 2019-11-20 at 6 57 21 PM

Screen Shot 2019-11-20 at 6 57 28 PM

### CODE REVIEW

File: ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx [hunk 1]
```diff
@@ -33,26 +40,30 @@ export const ApplicationsSyncPanel = ({show, apps, hide}: {show: boolean; apps:
...
                             }
-                            hide();
+                            setProgress({percentage: 100, title: 'Complete'});
                         }}
                         getApi={setForm}>
                         {formApi => (
                             <React.Fragment>
                                 <div className='argo-form-row'>
                                     <h4>Sync app(s)</h4>
+                                    {progress !== null && <ProgressPopup onClose={() => setProgress(null)} percentage={progress.percentage} title={progress.title} />}
```
Comment (alexmt) [inline ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx]: What happens if user close progess popup in a middle of syncing operation? Once the next app sync starts progress popup will be shown again?

**Review by alexmt — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx (hunk 1 above)
Comment (alexec) [reply ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx]: You can't. The pop-up is modal. Nor can you close the pop-up. You can close the window, which aborts the sync.

Comment (alexec) [reply ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx]: Question: do we need a better way to abort?

**Review by alexmt — APPROVED** [review]

File: ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx (hunk 1 above)
Comment (alexmt) [reply ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx]: I think it is good idea. Does not have to be part of the same PR though.

**Pull request merged**